addons/MountRoulette/MountRoulette.lua

```diff
@@ -0,0 +1,79 @@
+_addon.name    = 'Mount Roulette'
+_addon.author  = 'Dean James (Xurion of Bismarck)'
+_addon.version = '3.0.0'
+_addon.commands = {'mountroulette', 'mr'}
+
+require('lists')
+resources = require('resources')
+
+math.randomseed(os.time())
+
+allowed_mounts = L{}
+possible_mounts = L{}
+for _, mount in ipairs(resources.mounts) do
+    possible_mounts:append(mount.name:lower())
+end
+
+function update_allowed_mounts()
+    local kis = windower.ffxi.get_key_items()
+
+    for _, id in ipairs(kis) do
+        local ki = resources.key_items[id]
+        if ki.category == 'Mounts' and ki.name ~= "trainer's whistle" then -- Don't care about the quest KI
+            local mount_index = possible_mounts:find(function(possible_mount)
+                return windower.wc_match(ki.name:lower(), '♪' .. possible_mount .. '*')
+            end)
+            local mount = possible_mounts[mount_index]
+
+            -- Add this to allowed mounts if it is not already there
+            if not allowed_mounts:contains(mount) then
+                allowed_mounts:append(mount)
+            end
+        end
+    end
+end
+
+windower.register_event('load', function()
+    update_allowed_mounts()
+end)
+
+windower.register_event('incoming chunk', function(id)
+    if id == 0x055 then --ki update
+        update_allowed_mounts()
+    end
+end)
+
+windower.register_event('addon command', function(command)
```

Comment:
While this pattern by itself is nice for more complicated command handlers I feel it's unnecessarily complex here. I'd just make `//mr` run the roulette and remove the help text and all the boilerplate surrounding it, as well as the help message for it. I don't see a benefit to having this setup,

Reply:
No worries - this is the boilerplate left over from  removing the blacklist stuff. I'll cut this out as well.